pkg/build/apiserver/admission/secretinjector/admission.go

```diff
@@ -93,7 +104,13 @@ func (si *secretInjector) Admit(attr admission.Attributes) (err error) {
 	if match := urlpattern.Match(patterns, url); match != nil {
 		secretName := match.Cookie.(string)
 		glog.V(4).Infof(`secretinjector: matched secret "%s/%s" to buildconfig "%s"`, namespace, secretName, bc.GetName())
-		bc.Spec.Source.SourceSecret = &api.LocalObjectReference{Name: secretName}
+		if bc.Spec.Source.SourceSecret == nil || bc.Spec.Source.SourceSecret.Name != secretName {
```

Comment:
Pretty sure this method returns early if `bc.Spec.Source.SourceSecret != nil`, so the first predicate is extraneous

Reply:
and if it's always nil, then the second predicate will always npe.

I'm not seeing a job here for validation admission in this plugin.

Reply:
The func returns early on non-nil.

But on nil we still have to check whether `urlpattern.Match(patterns, url)` matches and then return the error during validation (a webhook could have reset the secret to nil).

The more fundamental question is whether we want to re-check at all in injection admission plugins. They are not security critical.